File: app/src/main/java/de/ironjan/mensaupb/menus_ui/MenuDetails.java
package de.ironjan.mensaupb.menus_ui;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.app.NavUtils;
import android.support.v7.app.ActionBarActivity;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Extra;
import org.androidannotations.annotations.OptionsItem;

import de.ironjan.mensaupb.R;

/**
 * Wrapping activity for {@link MenuDetailFragment}.
 */
@SuppressLint("Registered")
@EActivity(R.layout.activity_menu_details)
public class MenuDetails extends ActionBarActivity {
    @SuppressWarnings("WeakerAccess")
    @Extra(value = MenuDetailFragment.ARG_ID)
    long menuId;
    private MenuDetailFragment mFragment;

    @AfterViews
    void bindFragment() {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        mFragment = MenuDetailFragment.newInstance(menuId);
        ft.replace(R.id.fragmentMenuDetails, mFragment, "mFragment");
        ft.commit();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @OptionsItem(android.R.id.home)
    void navUp() {
        Intent intent = NavUtils.getParentActivityIntent(this);
        NavUtils.navigateUpTo(this, intent);
    }
}

File: app/src/main/java/de/ironjan/mensaupb/menus_ui/MenuDetails.kt
package de.ironjan.mensaupb.menus_ui

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v4.app.NavUtils
import android.support.v7.app.AppCompatActivity
import de.ironjan.mensaupb.R
import de.ironjan.mensaupb.api.model.Menu
import de.ironjan.mensaupb.menus_ui.MenuDetailFragment.Companion
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EActivity
import org.androidannotations.annotations.Extra
import org.androidannotations.annotations.OptionsItem

/**
 * Wrapping activity for [MenuDetailFragment].
 */
@SuppressLint("Registered")
@EActivity(R.layout.activity_menu_details)
open class MenuDetails : AppCompatActivity() {
    @Extra(value = MenuDetailFragment.PARCEL_MENU)
    internal lateinit var menu: Menu

    private var mFragment: MenuDetailFragment? = null

    @AfterViews
    internal fun bindFragment() {
        val ft = supportFragmentManager.beginTransaction()
        mFragment = Companion.newInstance(menu)
        ft.replace(R.id.fragmentMenuDetails, mFragment!!, "mFragment")
        ft.commit()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    @OptionsItem(android.R.id.home)
    internal fun navUp() {
        val intent = NavUtils.getParentActivityIntent(this)
        NavUtils.navigateUpTo(this, intent!!)
    }
}
